### ISSUE TITLE
ceph osd <df|stats|pool stats|perf..> hang without any visible error after node restart

### ISSUE THREAD

**Issue by bugre** [rook/rook#8980]
**Is this a bug report or feature request?**
* Bug Report. (not sure it's a bug)

**Deviation from expected behavior:**
rook ceph cluster, deployed using helm. Cluster has 6 nodes with 3 OSD's each. 

Cluster is providing service as expected, but osd monitoring/troubleshooting commands like (executed from within the toolbox pod):
* ceph osd pool stats
* ceph osd df
* ceph osd perf

hang on execution, without any logs/error messages. Also can't find any apparent error on logs
The strange thing is that commands like:
* ceph df
* ceph osd tree
* ceph osd pool ls detail
all work fine.

**Expected behavior:**

Commands should return the expected information.

**How to reproduce it (minimal and precise):**

This started happening after a node of the cluster had to be restarted because of hardware issues.

Unfortunately i have no spare hw available to create a new cluster, and the cluster having the issue is a production cluster so experimenting is somewhat restricted.

Before restart i've used :
```
  ceph osd set noout
  ceph osd set norebalance
```
after restart did the "unset" and cluster did sync data and continued providing service as expected.

I've restarted the toolbox, operator and mgr of the cluster, as a trial/error , but didn't have any improvement after those actions. The  'ceph osd <df|pool stats|perf...> ' commands still hang the same way.

Reviewing logs from mgr, mon-a, operator don't show (at least not that i detect) clear error indicative. 

On the ceph webgui, the OSD stat's are shown / updated as expected.

I'm looking for recommendation what/where to look with more attention to try find out what is happening.sl

Thanks for any help/tip.

**Environment**:
* OS (e.g. from /etc/os-release): Flatcar Container Linux by Kinvolk 2605.5.0 (Oklo)"
* Kernel (e.g. `uname -a`): Linux xxxx 5.4.66-flatcar #1 SMP Fri Sep 18 15:21:01 -00 2020 x86_64 Intel(R) Xeon(R) Silver 4214 CPU @ 2.20GHz GenuineIntel GNU/Linux
* Cloud provider or hardware configuration: n/a
* Rook version (use `rook version` inside of a Rook Pod): 
** rook: v1.4.4
**go: go1.13.8
* Storage backend version (e.g. for ceph do `ceph -v`):ceph version 15.2.4 (7447c15c6ff58d7fce91843b705a268a1917325c) octopus (stable)
* Kubernetes version (use `kubectl version`): v1.16.2
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): n/a
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest/ceph-toolbox.html)): HEALTH_OK

**Comment by bugre** [comment]
After your previous message @travisn ,  i went to investigate more on PG's, as i actually never used 'ceph' cli to do anything PG related. I've always used the ceph webgui for that.

![image](https://user-images.githubusercontent.com/6900672/137382507-fe2aafb5-f27d-4a85-8c36-b99eaaeb239d.png)

and the `ceph -s` cli output, stating the PG status.


So I've tried some of the PG commands on the ceph cli and those also are hanging without any return/message nor indication of error. 
* ceph pg getma
* ceph pg stat
* ceph pg dump all
* ceph pg ls-by-osd 1
* ... all freeze without any output/message executing the commands inside the toolbox pod.


But for example the command: 
* ceph osd pool get xxx-ceph-fs-data0 pg_num 
does return correctly the same value 32 as before the node was powercycled.



Thanks for any additional checking i could do.
-wm

**Comment by bugre** [comment]
Hi ,

Finally found out what was causing the strange behaviour, for the *tools* pod. 

Our k8s cluster has nodes that are not part of the CEPH cluster (they have no OSD's nor do they have a specific NIC / VLAN for CEPH data traffic), but all are part of the same k8s cluster.

Didn't expect the *tools* pod to need direct access to the ceph data network, and the reason was that (after reviewing logs i confirmed it) the tools pod was running on the node that was shutdown for maintenance and so the pod got relocated to another node, by chance :( the selected one by k8s was one of the nodes that is not part of the ceph nodes (no osd, nor ceph-data network)) Everything still worked as expected but certain commands (as described above) stopped working.

Found the problem, stracing the 'ceph' execution, and there it hung forever trying to connect to some of the IP's of the ceph data network (why i still don't know). 

I then edited the deployment adding a nodeSelector / rollout and the pod got recreated on a node that was part of the ceph nodes, and voylá, everything was working as before.


Question, is this the expected behaviour? Does the tools pod not talk over the regular k8s network with the other ceph pods to get its job done?

We deploy our ceph using helm chart, with some minor customizations, for specifically the restriction on what nodes the pods should be allocated, but for the tools deployment, we haven't could a easy way of doing it. For now, after cluster is running we've edited the deployment manually adding a nodeSelector, but there is probably a better way of doing it.

Thanks
